app/components/provider_interface/organisation_list_component.rb

```diff
@@ -0,0 +1,22 @@
+module ProviderInterface
+  class OrganisationListComponent < ViewComponent::Base
+    attr_accessor :training_permissions, :ratifying_permissions, :current_provider_user
+
+    def initialize(provider:,  current_provider_user:)
+      @current_provider_user = current_provider_user
+      @training_permissions = ProviderRelationshipPermissions.where(training_provider: provider)
```

Comment:
The main change we have done here is move the queries into a component. The core of the queries themselves have not changed:

We are using a `provider` instead of using `params[:id]`. I think because the query is broken down into an `or` the first with `.where(training_provider: provider)` and the second with `.where(training_provider: all_manageable_providers, ratifying_provider: provider)` this might have caused some confusion?